src/GitVersion.Core/Core/RepositoryStore.cs

```diff
@@ -239,52 +239,58 @@ public IEnumerable<IBranch> ExcludingBranches(IEnumerable<IBranch> branchesToExc
         // TODO Should we cache this?
         public IEnumerable<IBranch> GetBranchesContainingCommit(ICommit commit, IEnumerable<IBranch> branches = null, bool onlyTrackedBranches = false)
         {
-            branches ??= repository.Branches.ToList();
-            static bool IncludeTrackedBranches(IBranch branch, bool includeOnlyTracked) => includeOnlyTracked && branch.IsTracking || !includeOnlyTracked;
-
             if (commit == null)
             {
                 throw new ArgumentNullException(nameof(commit));
             }
 
-            using (log.IndentLog($"Getting branches containing the commit '{commit.Id}'."))
+
+            branches ??= repository.Branches.ToList();
+            static bool IncludeTrackedBranches(IBranch branch, bool includeOnlyTracked) => includeOnlyTracked && branch.IsTracking || !includeOnlyTracked;
+
+            return Impl();
+
+            IEnumerable<IBranch> Impl()
```

Comment:
Do you want to include a comment with the reasoning of the change? Do you want to add any specific test?

Reply:
>Let me know what's your preference for the name of the local function.

How about `InnerGetBranchesContainingCommit`?

```suggestion
            IEnumerable<IBranch> InnerGetBranchesContainingCommit()
```

> Do you want to include a comment with the reasoning of the change?

Yes, that would be nice.

> Do you want to add any specific test?

Yes, please.